client/session.go

```diff
@@ -1397,22 +1429,28 @@ func (s *session) streamBlocksMetadataFromPeers(
 	)
 
 	pending = int64(len(peers))
-	m.metadataFetches.Update(float64(pending))
+	progress.metadataFetches.Update(float64(pending))
 	for _, peer := range peers {
 		peer := peer
 
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			err := s.streamBlocksMetadataFromPeer(namespace, shard, peer,
-				start, end, ch, m)
+			var err error
+			if version == FetchBlocksMetadataEndpointV2 {
+				err = s.streamBlocksMetadataFromPeerV2(
+					namespace, shard, peer, start, end, metadataCh, progress)
+			} else {
+				err = s.streamBlocksMetadataFromPeer(
+					namespace, shard, peer, start, end, metadataCh, progress)
+			}
```

Comment:
Can we do a switch here too to verify that a valid support one was set?  i.e. switch V1, V2 and default to setting an error.

Reply:
Sure added the switch / default case. I can't really return the err since its an anonymous goroutine, but I can handle it the way we handle the other errs. Should be sufficient considering this should NeverHappenAnywaysTM